### DISCUSSION TITLE
aria2: Download links inside txt file

### DISCUSSION THREAD

**Discussion by bodysoda69** [anasty17/mirror-leech-telegram-bot#896] [Ideas]
Hey, I'm aware of the multi download option in the repo. Suppose if I need to download 50+ links and upload them to the same directory, multi option will spam messages and might cause floodwait errors too. 

So is it possible to add download from a input-file?
Reference: https://github.com/pawamoy/aria2p/blob/930f709e1d78f133e8cbd717ba4c7b9f761244ee/src/aria2p/api.py#L893

Hope this helps!

**Comment by anasty17** [comment]
Multi link will not spam or flood unless u changed the sleep time!

**Comment by anasty17** [comment]
This option will be used same as multi link. The only difference is that this function can return the list of download links then we should use the multi link functionality.

**Comment by bodysoda69** [comment]
Yeah, Multi link is good.

I'm asking for a new feature, not a replacement of multi link. If multi option is used for 50+ links it'll create 50+ tasks and all tasks upload to the parent_id.

In input file method, maybe you could download all the links one by one in the same task and upload all of them to a base folder.
Like one file completed and the another one starts. After all downloads are completed, the base folder gets uploaded.

This would be a game changer, since there are no bots that could download the links inside a file.

**Reply by anasty17** [reply]
Ur maybe is wrong. It doesn't work like this.

**Reply by bodysoda69** [reply]
https://github.com/Loli-Killer/python-aria-mirror-bot/blob/master/bot/helper/mirror_utils/download_utils/aria2_download.py

Above code allows to download multiple links in single task as one after the other and uploads them as a folder to drive. 

I know this is nothing related to the input-file idea, but just wanted to let you know instead of 50+ tasks it's possible to download all in a single task.

**Reply by anasty17** [reply]
I know it's possible! Already in todo list. But not actually like what u said. All links will be in same directory but all of them have separated task ( for unzip tg splits mainly) and upload will be after all downloads finish. This not worth all this coding! I will not add feature for few people need or silly things for abuse. In short stop abuse! Download only ur needs!

**Comment by anasty17** [comment]
When i add feature, then it will be for useful purpose. If u are abusers u shouldn't ask for anything.  U already have status pages and multi link enjoy! No more shit.
